src/maximal_cliques_template.h

```diff
@@ -100,6 +104,39 @@
 } while (0)
 #endif
 
+#ifdef IGRAPH_MC_NUMBER
+#define RESTYPE igraph_integer_t *res
+#define RESNAME res
+#define SUFFIX _number
+#define PREFIX i_
+#define RECORD do { \
+  if (*res < clsize) \
+    *res = clsize; \
+  } while (0)
+#define FINALLY *res=0;
+#define FOR_LOOP_OVER_VERTICES for (i=0; i<no_of_nodes; i++) {
+#define FOR_LOOP_OVER_VERTICES_PREPARE
+#endif
+
+#ifdef IGRAPH_MC_SINGLE_LARGEST
+#define RESTYPE igraph_vector_t *res
+#define RESNAME res
+#define SUFFIX _single_largest
+#define PREFIX i_
+#define RECORD do { \
+  if (igraph_vector_size(res) < clsize){ \
+    igraph_vector_init(res, clsize); \
+    int j; \
+    for (j=0; j<clsize; j++) { \
+      VECTOR(*res)[j] = VECTOR(*R)[j]; \
+    } \
+  } \
+  } while (0)
+#define FINALLY igraph_vector_init(res,0);
```

Comment:
Shouldn't this be `igraph_vector_clear` instead?  igraph functions typically expect already initialized vectors, which are then resized as necessary.  This way *both* allocation and destruction are left to the user (and not igraph), which avoids surprises.